### PR TITLE
Allow multiple colors on HL get and summary

### PR DESCRIPTION
**Pull request by jpslav**
Allow multiple colors on HL get and summary

### CODE REVIEW

File: app/models/highlight.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ class Highlight < ApplicationRecord
 
   scope :by_source_type, ->(source_type) { where(source_type: source_type) }
   scope :by_color, ->(color) { where(color: color) }
+  scope :by_colors, ->(colors) { where(color: colors) }
```
Comment (steveburkett) [inline app/models/highlight.rb:31]: do we still need the by_color scope?

File: app/controllers/api/v0/highlights_swagger.rb [hunk 2]
```diff
@@ -278,12 +278,16 @@ class Api::V0::HighlightsSwagger
...
-        key :type, :string
+        key :type, :array
+        key :collectionFormat, :csv
         key :required, false
-        key :enum, VALID_HIGHLIGHT_COLORS
-        key :description, 'Limits results to this highlight color.'
+        key :description, 'Limits results to these highlight colors.'
+        items do
+          key :type, :string
+          key :enum, VALID_HIGHLIGHT_COLORS
```
Comment (jpslav) [inline app/controllers/api/v0/highlights_swagger.rb:289]: our version of swagger codegen does not generate validations for enums in an array, which is why I had to write them separately

File: spec/requests/api/v0/highlights_controller_spec.rb [hunk 3]
```diff
@@ -101,6 +101,17 @@
           end
         end
 
+        context 'colors passed as comma-separated values in the query parameters' do
+          it 'handles them fine' do
+            get "/api/v0/highlights?source_type=openstax_page&scope_id=#{scope_1_id}&" \
+                "colors=yellow,pink"
```
Comment (steveburkett) [inline spec/requests/api/v0/highlights_controller_spec.rb:107]: does this need a test that tests that this test is not a false positive?  (e.g., testing for just yellow returns a subset of the yellow, pink?

File: spec/requests/api/v0/highlights_controller_spec.rb [hunk 4]
```diff
@@ -480,6 +499,27 @@
           highlight2.source_id.to_sym => 1,
         })
       end
+
+      context 'colors passed as comma-separated values in the query parameters' do
+        it 'handles them fine' do
+          get "/api/v0/highlights/summary?source_type=openstax_page&scope_id=#{scope_1_id}&" \
+              "colors=yellow,pink"
```
Comment (steveburkett) [inline spec/requests/api/v0/highlights_controller_spec.rb:506]: we note that the arrays are passed in as csv, and not as standard URL array syntax somewhere right?

File: spec/requests/api/v0/highlights_controller_spec.rb (hunk 3 above)
Comment (jpslav) [reply spec/requests/api/v0/highlights_controller_spec.rb:107]: This test already excludes the new `"green"` highlight

File: app/models/highlight.rb (hunk 1 above)
Comment (jpslav) [reply app/models/highlight.rb:31]: Maybe not but I think we could leave it.

File: spec/requests/api/v0/highlights_controller_spec.rb (hunk 4 above)
Comment (jpslav) [reply spec/requests/api/v0/highlights_controller_spec.rb:506]: yeah, it is `key :collectionFormat, :csv` in the swagger

**Review by steveburkett — APPROVED** [review]

**Pull request merged**